File: generateclasses/Matiere.java
package entite;

public class Matiere{
    int idmatiere; 
    String nommatiere; 
    int idunite; 

    public Matiere(){ }
    
    public int getIdmatiere(){
        return this.idmatiere;
    }
    public void setIdmatiere(int idmatiere){
        this.idmatiere=idmatiere;
    }
    public String getNommatiere(){
        return this.nommatiere;
    }
    public void setNommatiere(String nommatiere){
        this.nommatiere=nommatiere;
    }
    public int getIdunite(){
        return this.idunite;
    }
    public void setIdunite(int idunite){
        this.idunite=idunite;
    }

}

File: generateclasses/Matiere.cs
namespace entite;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("matiere")]
public class Matiere : Motherobj<Matiere>  {
    [Key]
    [Column("idmatiere")]
    public int idmatiere {get; set;} 
    [Column("nommatiere")]
    public string nommatiere {get; set;} 
    [Column("idunite")]
    public int idunite {get; set;} 

    public Matiere(){ }
    

}
